test/e2e/tuning_options_test.go

```diff
@@ -320,3 +327,195 @@ func TestHAProxyTimeoutsRejection(t *testing.T) {
 		}
 	}
 }
+
+// TestCookieLen verifies that when logging is enabled and a cookie capture is specified, any captured cookies are
+// truncated at the expected max length.
+func TestCookieLen(t *testing.T) {
+	testCases := []struct {
+		name      string
+		maxLength int
+	}{
+		{"63 bytes", 63},
+		{"64 bytes", 64},
+		{"128 bytes", 128},
+		{"1024 bytes", 1024},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			maxLength := testCase.maxLength
+			t.Parallel()
+			icName := types.NamespacedName{
+				Namespace: operatorNamespace,
+				Name:      names.SimpleNameGenerator.GenerateName("test-cookielen-"),
+			}
+			domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+			ic := newPrivateController(icName, domain)
```

Comment:
Is there a reason you need to create a new ic for each test case?

Reply:
The cases run in parallel, so unless I'm mistaken, they each need their own ingress controller.

Reply:
I was thinking - create one ic, and create one pod per test case.  That shouldn't cause any conflicts that I can think of.  I'm a little concerned that there could be resource failures or timeouts creating so many controllers.

Reply:
I don't think one ic would work; Each test case sets the ic's `spec.logging.access.httpCaptureCookies.maxLength` field to a different value, and you can only set one cookie to capture per ic.

Reply:
Normally we run each test in parallel, but here you are running each test case in parallel.  Did you get advice to do it this way?  I don't see the advantages, but do see the aforementioned potential disadvantage in that resource use could be compromised by running 4 private ics in parallel in ci.

Could you at least add a comment and note that it might cause an issue and point out how to fix that issue (don't run in parallel and use one private ic).

Reply:
We create many ingresscontoller's in the e2e tests and creating 4 more shouldn't be a problem (until some limit proves me wrong). 

If we created 4 new discrete TestFeature functions, each creating their own private IC, to accommodate and test new features over subsequent OCP releases would you have the same concerns? 

w.r.t running each test case in parallel--that seems fine to me. It's really no different to creating 4 new TestX functions that use `t.Parallel`.